signer/apk/apk.go

```diff
@@ -80,6 +81,19 @@ func (s *APKSigner) Config() signer.Configuration {
 //
 // This implements apksigning v1, aka jarsigner. apksigning v2 is not supported.
 func (s *APKSigner) SignFile(input []byte, options interface{}) (signer.SignedFile, error) {
+	opt, err := GetOptions(options)
+	if err != nil {
+		return nil, errors.Wrap(err, "apk: cannot get options")
+	}
+	if opt.ZIP == "" {
+		// default to compressing all files
+		opt.ZIP = "all"
+	}
+	err = validateZIPOption(opt.ZIP)
+	if err != nil {
+		return nil, errors.Wrap(err, "apk: got invalid ZIP option")
+	}
```

Comment:
nit: maybe these 8 lines should go into GetOptions?

Reply:
So I was doing that originally, but had a lot of trouble testing `GetOptions` directly (was getting "cannot unmarshal string into go value of type apk.Options") and couldn't find anywhere else that test `GetOptions` to copy either.